components/style/stylesheet_set.rs

```diff
@@ -0,0 +1,105 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+//! A centralized set of stylesheets for a document.
+
+use arc_ptr_eq;
+use std::sync::Arc;
+use stylesheets::Stylesheet;
+
+/// The set of stylesheets effective for a given document.
+pub struct StylesheetSet {
+    /// The actual list of all the stylesheets that apply to the given document.
+    ///
+    /// This is only a list of top-level stylesheets, and as such it doesn't
+    /// include recursive `@import` rules.
+    stylesheets: Vec<Arc<Stylesheet>>,
+
+    /// Whether the stylesheets list above has changed since the last restyle.
+    dirty: bool,
+
+    /// Has author style been disabled?
+    author_style_disabled: bool,
+}
+
+impl StylesheetSet {
+    /// Create a new empty StylesheetSet.
+    pub fn new() -> Self {
+        StylesheetSet {
+            stylesheets: vec![],
+            dirty: false,
```

Comment:
Note that this is different from the old initial value, though not sure if it matters.